### PR TITLE
Add support for ATC_MiThermometer

### LINKED ISSUE #877: Add support for ATC_MiThermometer
**Describe the problem you have/What new integration you would like**

I'd like to see ATC_MiThermometer supported in esphome.

**Please describe your use case for this integration and alternatives you've tried:**

The original firmware is supported but it requires complicated bind key generation and only sends the data once in 10 minutes.
The opensource firmware can be programmed to any interval.

**Additional context**

This has already been discussed a bit in closed issue #552 . I'm opening this one to not spoil the original issue.
All protocol info can be found here: https://github.com/atc1441/ATC_MiThermometer

### PR DESCRIPTION
**Pull request by ahpohl**
## Description:

This patch adds support for the [ATC MiThermometer](https://github.com/atc1441/ATC_MiThermometer) custom firmware, which replaces the stock firmware in Xiaomi LYWSD03MMC Temperature/Humitity sensors. The main advantage of the custom firmware compared to the Xiaomi stock firmware is that the payload is not encrypted and there is no [bindkey](https://esphome.io/components/sensor/xiaomi_ble.html#obtaining-the-bindkey) necessary.

**Note:** This patch (i.e. the new `atc_mithermometer` component) specifically requires the custom firmware to be operated in "custom" advertising mode. The firmware also has a "Mi Like" advertising mode, which is already fully supported by the `xiaomi_lywsd03mmc` component (with a dummy bindkey still necessary in the config).

telink_flasher

**Related issue (if applicable):** fixes https://github.com/esphome/feature-requests/issues/877

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#777

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/atc_mithermometer/sensor.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import sensor, esp32_ble_tracker
+from esphome.const import CONF_BATTERY_LEVEL, CONF_HUMIDITY, CONF_MAC_ADDRESS, CONF_TEMPERATURE, \
+    UNIT_CELSIUS, ICON_THERMOMETER, UNIT_PERCENT, ICON_WATER_PERCENT, ICON_BATTERY, CONF_ID
+
```
Comment (glmnet) [inline esphome/components/atc_mithermometer/sensor.py:6]:
```suggestion

CODEOWNERS = ['@ahpohl']
```

File: esphome/components/atc_mithermometer/atc_mithermometer.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  bool parse_device(const esp32_ble_tracker::ESPBTDevice &device) override;
+  void dump_config() override;
+  float get_setup_priority() const override { return setup_priority::DATA; }
+  void set_temperature(sensor::Sensor *temperature) { temperature_ = temperature; }
+  void set_humidity(sensor::Sensor *humidity) { humidity_ = humidity; }
+  void set_battery_level(sensor::Sensor *battery_level) { battery_level_ = battery_level; }
+
+  optional<ParseResult> parse_header(const esp32_ble_tracker::ServiceData &service_data);
+  bool parse_message(const std::vector<uint8_t> &message, ParseResult &result);
+  bool report_results(const optional<ParseResult> &result, const std::string &address);
```
Comment (glmnet) [inline esphome/components/atc_mithermometer/atc_mithermometer.h:33]: I would mark those protected, but it's ok too

**Review by glmnet — COMMENTED** [review]
It'd be nice if you add CODEOWNERS to all your contributions, that will ease issue handling, thanks

File: requirements.txt [hunk 3]
```diff
@@ -1,7 +1,7 @@
-voluptuous==0.11.7
+voluptuous==0.12.0
 PyYAML==5.3.1
 paho-mqtt==1.5.1
-colorlog==4.2.1
+colorlog==4.4.0
 tornado==6.0.4
 protobuf==3.13.0
 tzlocal==2.1
```
Comment (jesserockz) [inline requirements.txt:7]: Please don't bump the dependencies if it is unrelated to your changes in the PR

File: esphome/components/atc_mithermometer/sensor.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        cg.add(var.set_humidity(sens))
+    if CONF_BATTERY_LEVEL in config:
+        sens = yield sensor.new_sensor(config[CONF_BATTERY_LEVEL])
+        cg.add(var.set_battery_level(sens))
+    if CONF_BATTERY_VOLTAGE in config:
+        sens = yield sensor.new_sensor(config[CONF_BATTERY_VOLTAGE])
+        cg.add(var.set_battery_voltage(sens))
+
+
+CODEOWNERS = ['@ahpohl']
```
Comment (jesserockz) [inline esphome/components/atc_mithermometer/sensor.py:46]: Could you move this to near the top of the file before `DEPENDENCIES` for consistency with other files and easy to read/find. Thanks

File: esphome/components/xiaomi_lywsd03mmc/sensor.py [hunk 5]
```diff
@@ -40,3 +40,6 @@ def to_code(config):
     if CONF_BATTERY_LEVEL in config:
         sens = yield sensor.new_sensor(config[CONF_BATTERY_LEVEL])
         cg.add(var.set_battery_level(sens))
+
+
+CODEOWNERS = ['@ahpohl']
```
Comment (jesserockz) [inline esphome/components/xiaomi_lywsd03mmc/sensor.py:45]: Could you move this to near the top of the file before `DEPENDENCIES` for consistency with other files and easy to read/find. Thanks

**Review by jesserockz — CHANGES_REQUESTED** [review]
Looking good, a few minor change please =)

**Review by jesserockz — APPROVED** [review]
Thanks

**Pull request merged**